src/common/promises/promise-factory.ts

```diff
@@ -25,3 +27,19 @@ export const createDefaultPromiseFactory = (): PromiseFactory => {
         timeout: createTimeout,
     };
 };
+
+export const createAlarmPromiseFactory = (alarmUtils: AlarmUtils): PromiseFactory => {
+    return {
+        timeout: <T>(promise: Promise<T>, delayInMilliseconds: number) => {
```

Comment:
nit: I think this would be more readable if we followed the pattern used for `createDefaultPromiseFactory` and separate the timeout function into a separate variable.